src/main/java/org/opensearch/knn/index/codec/transfer/VectorTransfer.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.transfer;
+
+import lombok.Data;
+import org.opensearch.knn.index.codec.util.SerializationMode;
+
+import java.io.ByteArrayInputStream;
+
+/**
+ * Abstract class to transfer vector value from Java to native memory
+ */
+@Data
+public abstract class VectorTransfer {
+    protected final long vectorsStreamingMemoryLimit;
+    protected long totalLiveDocs;
+    protected long vectorsPerTransfer;
+    protected long vectorAddress;
+    protected int dimension;
+
+    public VectorTransfer(final long vectorsStreamingMemoryLimit) {
+        this.vectorsStreamingMemoryLimit = vectorsStreamingMemoryLimit;
+        this.vectorsPerTransfer = Integer.MIN_VALUE;
+    }
+
+    /**
+     * Initialize the transfer
+     *
+     * @param totalLiveDocs total number of vectors to be transferred
+     */
+    abstract public void init(final long totalLiveDocs);
+
+    /**
+     * Transfer a single vector
+     *
+     * @param byteStream a vector in byte stream format
+     */
+    abstract public void transfer(final ByteArrayInputStream byteStream);
+
+    /**
+     * Close the transfer
+     */
+    abstract public void close();
```

Comment:
how we are ensuring that close method is always called?

Reply:
I thought it is caller's responsibility to call the close. Any idea to enforce it?

Reply:
Let me make it as closable class.

Reply:
I think I got your point. I think it should be refactored so that there is no dependency between methods.